File: src/main/java/superstitio/cards/maso/SkillCard/CutWrist.java
package superstitio.cards.maso.SkillCard;

import com.megacrit.cardcrawl.actions.common.LoseHPAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.maso.MasoCard;
import superstitio.delayHpLose.RemoveDelayHpLoseBlock;

public class CutWrist extends MasoCard {
    public static final String ID = DataManager.MakeTextID(CutWrist.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.COMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int MAGIC = 8;
    private static final int BLOCK = 18;
    private static final int UPGRADE_BLOCK = 4;

    public CutWrist() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC);
        this.setupBlock(BLOCK, UPGRADE_BLOCK, new RemoveDelayHpLoseBlock());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot(new LoseHPAction(AbstractDungeon.player, AbstractDungeon.player, this.magicNumber));
        addToBot_gainBlock();
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/maso/SkillCard/CutWrist.kt
package superstitio.cards.maso.SkillCard

import basemod.cardmods.RetainMod
import basemod.helpers.CardModifierManager
import com.megacrit.cardcrawl.actions.common.LoseHPAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.maso.MasoCard
import superstitio.delayHpLose.RemoveDelayHpLoseBlock

class CutWrist : MasoCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC)
        this.setupBlock(BLOCK, UPGRADE_BLOCK, RemoveDelayHpLoseBlock())
        CardModifierManager.addModifier(this, RetainMod())
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot(LoseHPAction(AbstractDungeon.player, AbstractDungeon.player, this.magicNumber))
        addToBot_gainBlock()
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(CutWrist::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.COMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 1
        private const val MAGIC = 8
        private const val BLOCK = 18
        private const val UPGRADE_BLOCK = 4
    }
}
